347.TopKFrequentElements.cpp

```diff
@@ -0,0 +1,42 @@
+class Solution
+{
+public:
+    // 347. Top K Frequent Elements: https://leetcode.com/problems/top-k-frequent-elements/
+    // Time Complexity: O(n)
+    // Space Complexity: O(n)
+
+    vector<int> topKFrequent(const vector<int> &nums, int k)
+    {
+        // Space Complexity: O(n)
+        unordered_map<int, int> data;
+
+        // Store all numbers in the dictionary and count how many
+        // Time Complexity: O(n)
+        for (const auto &num : nums)
+            data[num]++;
+
+        // Space Complexity: O(n)
+        vector<pair<int, int>> totals;
+
+        // Store the counts as the first value in the pair for sorting in priority_queue
+        // Time Complexity: O(n)
+        for (const auto &[key, value] : data)
+            totals.push_back({value, key});
+
+        // Time Complexity: O(n)
+        priority_queue<pair<int, int>> pq(totals.begin(), totals.end());
+
+        int i = 0;
+        vector<int> res(k, 0);
```

Comment:
optional: Prefer to avoid abbreviations for variable names if possible - `result` here would be more descriptive than `res` at the cost of three additional characters.

Reply:
addressed